sklearn/plot/_gridsearch_results.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from sklearn.plot import plot_heatmap
+
+
+def _plot_1D_results(cv_results, params, ax, xlabel, ylabel, title,
+                     fmt, xtickrotation):
+    import matplotlib.pyplot as plt
```

Comment:
yea agree.
i was suggesting to extend this behaviour to plot_confusion_matrix and other functions which might use plot_heatmap in future. if we make ax compulsory, we ensure axis is being checked closer to public layer.